batch/src/main/scala/uk/gov/ons/addressindex/models/EsDocument.scala

```diff
@@ -22,27 +24,23 @@ abstract class EsDocument {
                                   subBuildingName: String, buildingName: String, doubleDependentLocality: String,
                                   dependentLocality: String, postTown: String, postcode: String): String = {
 
-    val poBoxNumberEdit = if (poBoxNumber.isEmpty) "" else s"PO BOX ${poBoxNumber}"
-    val numberAndLetter = "\\d+[A-Z]".r
-
-    val trimmedBuildingNumber = buildingNumber.trim
-    val trimmedDependentThoroughfare = splitAndCapitalise(dependentThoroughfare)
-    val trimmedThoroughfare = splitAndCapitalise(thoroughfare)
-
-    val buildingNumberWithStreetName =
-      s"$trimmedBuildingNumber ${if (trimmedDependentThoroughfare.nonEmpty) s"$trimmedDependentThoroughfare, " else ""}$trimmedThoroughfare"
-
-    val departmentNameEdit = splitAndCapitalise(departmentName)
-    val organisationNameEdit = splitAndCapitalise(organisationName)
-    val subBuildingNameEdit = splitAndCapitalise(subBuildingName)
-    val buildingNameEdit = if (numberAndLetter.findFirstIn(buildingName.trim).isEmpty) splitAndCapitalise(buildingName) else buildingName
-    val doubleDependentLocalityEdit = splitAndCapitaliseTowns(doubleDependentLocality)
-    val dependentLocalityEdit = splitAndCapitaliseTowns(dependentLocality)
-    val postTownEdit = splitAndCapitaliseTowns(postTown)
-
-    Seq(departmentNameEdit, organisationNameEdit, subBuildingNameEdit, buildingNameEdit,
-      poBoxNumberEdit, buildingNumberWithStreetName, doubleDependentLocalityEdit, dependentLocalityEdit,
-      postTownEdit, postcode).map(_.trim).filter(_.nonEmpty).mkString(", ")
+    val thoroughfares = Seq(dependentThoroughfare, thoroughfare).map(normalize).map(strToOpt)
+    val premises = Seq(subBuildingName, buildingName, buildingNumber).map(normalize).map(strToOpt)
+    val poBox = strToOpt(poBoxNumber).map("PO BOX " + _)
+
+    // merge the first entry in thoroughfare, and the last entry in premises, if they exist
+    val premsAndThoroughfare = (premises, thoroughfares) match {
```

Comment:
This is nice and concise but could you name the internal variables something more human readable?